package.py

```diff
@@ -0,0 +1,911 @@
+#!/usr/bin/env python3
+import argparse
+import base64
+import logging
+import os
+import platform
+import shutil
+import stat
+import subprocess
+from pathlib import Path
+from typing import Any, Callable, Literal, Optional
+from zipfile import ZipFile
+
+import requests
+
+from packaging import version
+
+pyinstaller_version = os.environ.get('PYINSTALLER_VERSION', '5.1')
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s: %(message)s',
+)
+logger = logging.getLogger('package')
+
+
+def log_group(name: str) -> Callable:
+    def start_group(group_name: str) -> None:
+        if os.environ.get('CI'):
+            subprocess.call(f'echo ::group::"{group_name}"', shell=True)
+        else:
+            logger.info(f'\n\n-----{group_name}-----\n\n')
+
+    def end_group() -> None:
+        if os.environ.get('CI'):
+            subprocess.call('echo ::endgroup::', shell=True)
+        else:
+            logger.info('\n\n-----------------\n\n')
+
+    def decorate(fn: Callable) -> Callable:
+        def wrapper(*args: Any, **kwargs: Optional[Any]) -> None:
+            start_group(name)
+            fn(*args, **kwargs)
+            end_group()
+
+        return wrapper
+    return decorate
+
+
+class Downloader:
+    @staticmethod
+    def download(url: str) -> str:
+        filename = url.split('/')[-1]
+        with requests.get(url, stream=True) as r:
+            r.raise_for_status()
+            with open(filename, 'wb') as f:
+                for chunk in r.iter_content(chunk_size=8192):
+                    f.write(chunk)
+        return filename
+
+
+class Environment:
+    def __init__(self) -> None:
+        self.archs = [
+            'AMD64',  # Windows
+            'x86_64',
+            'aarch64',
+            'arm64',
+        ]
+
+        self.oses = [
+            'Darwin',
+            'Windows',
+            'Linux',
+        ]
+
+        self.arch = platform.machine()
+        self.os = platform.system()
+        self.target_arch = os.environ.get('MACOS_BUILD_ARCH', self.arch)
+
+        if self.is_mac():
+            os.environ.setdefault('ONEFILE', '0')
+
+        python_bin = 'python3'
+        if self.is_windows():
+            python_bin = 'python'
+
+        self.rotki_version = subprocess.check_output(
+            f'{python_bin} setup.py --version',
+            shell=True,
+            encoding='utf8',
+        ).strip()
```

Comment:
If for some reason you don't have the file you would get

```
python: can't open file
```
I  believe this is something you want to catch

Reply:
Why though, the script should run from the root of the repo. If the setup.py is missing you already did something that you should not.

Reply:
Exactly, so we can detect earlier possible errors